completions/dpkg

```diff
@@ -1,37 +1,62 @@
 # dpkg(1) and related commands completion                  -*- shell-script -*-
 
-if _comp_have_command grep-status; then
-    _comp_xfunc_dpkg_installed_packages()
-    {
-        grep-status -P -e "^${1-}" -a -FStatus 'ok installed' -n -s Package
-    }
-
-    _comp_xfunc_dpkg_purgeable_packages()
-    {
-        grep-status -P -e "^${1-}" -a -FStatus 'ok installed' -o -FStatus 'ok config-files' -n -s Package
-    }
-else
-    _comp_xfunc_dpkg_installed_packages()
-    {
+_comp_xfunc_dpkg_installed_packages()
+{
+    local pkgs
+    pkgs=$(
+        grep-status -P -e "^${cur-}" -a \
+            -FStatus 'ok installed' \
+            -n -s Package 2>/dev/null
+    ) || pkgs=$(
```

Comment:
We could check if the exit code is 127 (or 126) rather than fall back on all failures, but I suppose this would be fine.